src/ke.cpp

```diff
@@ -40,8 +98,17 @@ KeRaiseIrqlToDpcLevel()
 void
 KeLowerIrql(_In_ KIRQL new_irql)
 {
+    BOOL result;
+    if (_usersim_current_irql >= DISPATCH_LEVEL && new_irql < DISPATCH_LEVEL) {
+        uint32_t processor_index = KeGetCurrentProcessorNumberEx(NULL);
```

Comment:
nit: in future please use `nullptr` instead of `NULL` in cpp files.
```suggestion
        uint32_t processor_index = KeGetCurrentProcessorNumberEx(nullptr);
```